src/cz/crcs/ectester/standalone/libs/jni/NativeSignatureSpi.java

```diff
@@ -544,4 +546,91 @@ public MscngECDSAwithSHA512() {
             super("SHA512withECDSA");
         }
     }
+
+    public abstract static class Nettle extends SimpleSignatureSpi {
+        private String type;
+
+        public Nettle(String type) {
+            this.type = type;
+        }
+
+        @Override
+        byte[] sign(byte[] data, byte[] privKey, ECParameterSpec params) {
+            try {
+                AlgorithmParameters tmp = AlgorithmParameters.getInstance("EC");
+                tmp.init(params);
+                ECGenParameterSpec spec = tmp.getParameterSpec(ECGenParameterSpec.class);
+                switch (spec.getName()) {
```

Comment:
It would be cool if this kind of lookup wasn't here, it actually uses `SunEC` to lookup the curve

(see https://hg.openjdk.java.net/jdk/jdk/file/9b513dbd57a3/src/java.base/share/classes/sun/security/util/ECParameters.java#l206 and https://hg.openjdk.java.net/jdk/jdk/file/9b513dbd57a3/src/java.base/share/classes/sun/security/util/CurveDB.java#l86) but I can fix this later, if it works, don't break it. I will add a similar lookup method to `EC_Store`.